### PR TITLE
feat(design-system): Storybook-style Cypress component tests

### PR DESCRIPTION
**Pull request by tkaragianes**
## Summary
We plan to use Cypress for testing and documenting our component library. Cypress provides similar features as Storybook, and is extensible for future improvements.

### CODE REVIEW

File: packages/design-system/vue/cypress/support/component-index.html [hunk 1]
```diff
@@ -1,11 +1,11 @@
 <!DOCTYPE html>
-<html>
+<html class="dark">
 <head>
   <meta charset="utf-8">
   <meta http-equiv="X-UA-Compatible" content="IE=edge">
   <meta name="viewport" content="width=device-width,initial-scale=1.0">
   <title>Components App</title>
-
+  <link rel="stylesheet" href="/__cypress/src/src/assets/style.css">
```
Comment (luxaritas) [inline packages/design-system/vue/cypress/support/component-index.html:8]: Why is this necessary when we already do this here? https://github.com/eternagame/eterna-next/blob/main/packages/design-system/vue/cypress/support/component.ts#L23

File: packages/design-system/vue/cypress/support/component.ts [hunk 2]
```diff
@@ -39,7 +41,12 @@ declare global {
   }
 }
 
-Cypress.Commands.add('mount', mount);
+// Use this to mount without ActionBar wrapper component
+// Cypress.Commands.add('mount', mount);
+
+// Augment mount command with an ActionBar wrapper component
+// The ActionBar is where we place app-level settings like theme
+Cypress.Commands.add('mount', (component, ...args) => mount(() => h(ActionBar, {}, () => h(component, args[0].props, args[0].slots)), ...args));
```
Comment (luxaritas) [inline packages/design-system/vue/cypress/support/component.ts:49]: Presumably the props and slots should be omitted from the args passed to mount?

Also this is getting a bit dense and hard to parse, I'd suggest splitting this to be multiline

File: packages/design-system/vue/cypress/support/component-index.html [hunk 3]
```diff
@@ -1,11 +1,11 @@
 <!DOCTYPE html>
-<html>
+<html class="dark">
```
Comment (luxaritas) [inline packages/design-system/vue/cypress/support/component-index.html:2]: Could the action bar handle setting the default theme instead?

**Review by luxaritas — COMMENTED** [review]
FWIW, this will probably also be needed in the frontend-web package, right?

File: packages/design-system/vue/cypress/support/ActionBar.vue [hunk 4]
```diff
@@ -0,0 +1,47 @@
+<template>
+  <header class="actionBar">
+    <div class="actionBar-themeSelect">
+      <span>Theme: </span>
+      <select
+        id="themeSwitcher"
+        name="themeSwitcher"
+        @:change="changeTheme"
```
Comment (luxaritas) [inline packages/design-system/vue/cypress/support/ActionBar.vue:8]: `@:change` or `@change`?

File: packages/design-system/vue/cypress/support/ActionBar.vue [hunk 5]
```diff
@@ -0,0 +1,47 @@
+<template>
+  <header class="actionBar">
+    <div class="actionBar-themeSelect">
+      <span>Theme: </span>
```
Comment (luxaritas) [inline packages/design-system/vue/cypress/support/ActionBar.vue:4]: Label?

File: packages/design-system/vue/cypress/support/ActionBar.vue (hunk 4 above)
Comment (tkaragianes) [reply packages/design-system/vue/cypress/support/ActionBar.vue:8]: Typo, should be @change.

File: packages/design-system/vue/cypress/support/component-index.html (hunk 1 above)
Comment (tkaragianes) [reply packages/design-system/vue/cypress/support/component-index.html:8]: Now I'm not sure why I added this, but you're right it's unnecessary.

File: packages/design-system/vue/cypress/support/ActionBar.vue [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  </header>
+  <main>
+    <slot />
+  </main>
+</template>
+
+<script setup lang="ts">
+function changeTheme(event: Event) {
+  const theme = (event.target as HTMLInputElement).value;
+  document.querySelector('html')?.setAttribute('class', theme);
```
Comment (luxaritas) [inline packages/design-system/vue/cypress/support/ActionBar.vue:27]: Out of scope for this PR, but I'm wondering if it would be better to define our variables on something like `.eds-theme-dark` instead of tying it to `:root` so we don't need to "break out" of our app to set the theme class

**Review by luxaritas — APPROVED** [review]
